src/session/sidebar/avatar.rs

```diff
@@ -159,36 +159,38 @@ impl Avatar {
 
     fn setup_is_online_binding(&self, user: &User) {
         if !matches!(user.type_().0, UserType::Regular) {
+            self.set_is_online(false);
             return;
         }
 
-        if let Some(session) = self.ancestor(Session::static_type()) {
-            let self_ = imp::Avatar::from_instance(self);
-
-            let me_expression =
-                gtk::PropertyExpression::new(Session::static_type(), gtk::NONE_EXPRESSION, "me");
-
-            let user_expression = gtk::ConstantExpression::new(user);
-            let interlocutor_status_expression = User::status_expression(&user_expression);
-
-            let user_id = user.id();
-            let interlocutor_is_online_expression = gtk::ClosureExpression::new(
-                move |args| {
-                    let maybe_me = args[1].get::<User>();
-                    maybe_me
-                        .map(|me| {
-                            let user_status = args[2].get::<BoxedUserStatus>().unwrap().0;
-                            matches!(user_status, UserStatus::Online(_)) && me.id() != user_id
-                        })
-                        .unwrap_or_default()
-                },
-                &[me_expression.upcast(), interlocutor_status_expression],
-            );
-
-            let is_online_binding =
-                interlocutor_is_online_expression.bind(self, "is-online", Some(&session));
-            self_.binding.replace(Some(is_online_binding));
-        }
+        // This should never panic as there must always be a `Session` as ancestor.
+        let session = self.ancestor(Session::static_type()).unwrap();
```

Comment:
Since with #166 we'll not have a `Session` widget anymore, I'd say to also change this to get the `Sidebar` instead and getting the `session()` property from there.